File: Matlab/incrementalSearch.m
function [answer,matrix] = incrementalSearch(f,x0,delta,niter)
fx0 =f(x0);
matrix=["iteration","x0","x1","fx0","fx1","fx0*fx1"];
if(fx0==0)
    answer=x0+" is a root";
else
    x1=x0+delta;
    counter=1;
    fx1=f(x1);
    while(fx0*fx1>0 && counter<niter)
        B=[counter,x0,x1,fx0,fx1,fx0*fx1];
        matrix=[matrix;B];
        x0=x1;
        fx0=fx1;
        x1=x0+delta;
        fx1=f(x1);
        counter=counter+1;
    end
    B=[counter,x0,x1,fx0,fx1,fx0*fx1];
    matrix=[matrix;B];
    if(fx1==0)
        answer=x1+" is a root";
    elseif(fx0*fx1<0)
        answer="There is at least one root between "+x0+" and "+x1;
    else
        answer="The method failed in "+niter+" iterations";
    end
end

File: NumericSquadProject/numericApp/methods/Roots/incrementalSearch.py
import sympy as sm

x = sm.symbols('x')

def incrementalSearch(f,xi,delta,niter):
    if(f.subs(x,xi)==0):
        return (str(xi) + " is a root")
        
    else:
        xf = xi + delta
        ite = 0
        while((float(f.subs(x,xi))*float(f.subs(x,xf)))>0 and ite < niter):
            xi = xf 
            xf = xf + delta
            ite += 1
            
        if((f.subs(x,xi)*f.subs(x,xf))==0):
            return (str(xf) + " is a root")
        
        elif((f.subs(x,xi)*f.subs(x,xf))<0):
            return ("Between " + str(xi) + " and " + str(xf) + " there is a root")
        
        else:
            return ("No root found")


# f = sm.exp(3*x-12) + x*sm.cos(3*x)-x**2+4
# xi = -10
# delta = 1
# niter = 10

# print(incrementalSearch(f,xi,delta,niter))